.github/workflows/test-and-release.yml

```diff
@@ -13,16 +13,24 @@ env:
 jobs:
   # unit tests
   units:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
+        with:
```

Comment:
By default, it will checkout `github.sha`, and for PR triggered builds, that will fetch the temporary merge commit SHA, instead of the source branch's head commit.